src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIInteract.java

```diff
@@ -80,13 +81,14 @@ protected final boolean mineBlock(@NotNull final BlockPos blockToMine)
      */
     protected final boolean mineBlock(@NotNull final BlockPos blockToMine, @NotNull final BlockPos safeStand)
     {
-        Block curBlock = world.getBlockState(blockToMine).getBlock();
+        IBlockState curBlockState = world.getBlockState(blockToMine);
+        Block curBlock = curBlockState.getBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'curBlock' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)